pkg/shipper/shipper_test.go

```diff
@@ -93,21 +87,35 @@ func TestShipper_UploadBlocks(t *testing.T) {
 
 		testutil.Ok(t, enc.Encode(&meta))
 
+		expBlocks[id] = struct{}{}
+
 		expFiles[id.String()+"/meta.json"] = buf.Bytes()
 		expFiles[id.String()+"/index"] = []byte("indexcontents")
 		expFiles[id.String()+"/chunks/0001"] = []byte("chunkcontents1")
 		expFiles[id.String()+"/chunks/0002"] = []byte("chunkcontents2")
-	}
 
-	testutil.Equals(t, timestamp.FromTime(now), gotMinTime)
+		// The shipper meta file should show all blocks as uploaded.
+		shipMeta, err := ReadMetaFile(dir)
```

Comment:
Can we add some random block in the middle after shipper already uploaded everything? to check if it will not reupload that?

Reply:
How do you mean? If we add a random block, it would upload it because it hasn't done so before. What we could do is delete one block in the bucket.

Reply:
sorry, got confused with the algorithm here. Yes, deleting would be nice to see if it got uploaded or not again.